src/main/java/com/p14n/postevent/PostgresDebeziumConnector.java

```diff
@@ -0,0 +1,54 @@
+package com.p14n.postevent;
+
+import io.debezium.engine.ChangeEvent;
+import io.debezium.engine.DebeziumEngine;
+import io.debezium.engine.format.Json;
+
+import java.io.IOException;
+import java.util.Properties;
+import java.util.function.Consumer;
+
+public class PostgresDebeziumConnector {
+    private final Properties properties;
+    private Consumer<ChangeEvent<String, String>> changeEventConsumer;
+    private DebeziumEngine<ChangeEvent<String, String>> engine;
+
+    public PostgresDebeziumConnector(Properties properties) {
+        this.properties = properties;
+    }
+
+    public void setChangeEventConsumer(Consumer<ChangeEvent<String, String>> consumer) {
+        this.changeEventConsumer = consumer;
+    }
+
+    public void start() {
+        if (changeEventConsumer == null) {
+            throw new IllegalStateException("Change event consumer must be set before starting the engine");
+        }
+
+        this.engine = DebeziumEngine.create(Json.class)
+                .using(properties)
+                .notifying(changeEventConsumer)
+                .build();
+
+        var thread = new Thread(() -> {
```

Comment:
instead of creating a thread directly, use an ExecutorService that is passed in as a parameter